### PR TITLE
Update various latex-lab modules

### PR DESCRIPTION
**Pull request by u-fischer**
This extends and changes various modules of latex-lab 

*  new latex-lab-title module:  adds support for \maketitle, uses the `\title` also for PDF-metadata, and -- if the UA-standard is used -- as window title
* new latex-lab-marginpar module. Support \marginpar (integrated into phase-III).
* latex-lab-math has been reworked in various places
       * sockets are used in various places
       * handling of the `\text` command has been corrected (it no longer produces broken PDF), 
       * support for reading mathml and attaching them as mathml-AF 
       
* latex-lab-table supports "layout tables". 
* latex-lab-firstaid: support for the `\author` command in amsart 
* Various bug fixes from tagging-project: 
   https://github.com/latex3/tagging-project/issues/29
   https://github.com/latex3/tagging-project/issues/34
   https://github.com/latex3/tagging-project/issues/31
   https://github.com/latex3/tagging-project/issues/50
   https://github.com/latex3/tagging-project/issues/55
   https://github.com/latex3/tagging-project/issues/57

* various documentation fixes

* updates tagpdf in the texmf tree

## Status of pull request
[x] Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [ ] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: required/latex-lab/latex-lab-math.dtx [hunk 1]
```diff
@@ -129,38 +164,274 @@
...
+%       \end{itemize}
+%   \end{itemize} 
+%    
+%  \subsection{Display math}
+%    
+%  \textit{to be written}
+%    
+%  \subsection{Associated Files}
+%  
+%  The current code allows to attach two type of associated files to the
```
Comment (davidcarlisle) [inline required/latex-lab/latex-lab-math.dtx:236]: %  The current code allows the attachment of two types of associated file to the

File: required/latex-lab/latex-lab-math.dtx [hunk 2]
```diff
@@ -129,38 +164,274 @@
...
+%  For a MathML representation a file with such representations must be provided. 
+%  If the equation is numbered the numbering should be part of the MathML as 
+%  the |Lbl| substructure is ignored if an MathML is used (see https://github.com/foxitsoftware/PDF_UA-2).
+%    
+%  The MathML representation is given in a special format. 
+%  It is meant to be a valid html file
+%  that can be viewed  in a browser.
+%  For this it can start with |<!DOCTYPE html><html>| and end with |</html>|
+%  It should have the extension \texttt{.html}. The \meta{mathml} content 
+%  is read with special catcodes, so can contain ambersands, hashes and comment chars.
```
Comment (davidcarlisle) [inline required/latex-lab/latex-lab-math.dtx:265]: and unmatched braces such as `<mo>{</mo>`

File: required/latex-lab/latex-lab-math.dtx [hunk 3]
```diff
@@ -129,38 +164,274 @@
...
+%  \cs{m@th} and if yes it doesn't call the processor before reinserting
+%  the content and perhaps adding tagging code.
+%  This method requires that the math can be grabbed (e.g. that the end dollar is visible)
+%  and that the \cs{m@th} is visible. It applies for example in \cs{@iiiparbox} where the
+%  code from |$\vcenter| to |\m@th$| is grabbed an put back. It does not work for
+%  example for |tabular| where the dollars and the \cs{m@th} token are spread around
+%  over three commands. |tabular| needs therefore manual intervention. 
+%  
+%  A look in the list of usages (in \texttt{usage-of-m@th.md}) justifies this approach.
+%  All usages are either no math at all, or related to small elements that probably
```
Comment (davidcarlisle) [inline required/latex-lab/latex-lab-math.dtx:379]: not math at all

File: required/latex-lab/latex-lab-amsmath.dtx [hunk 4]
```diff
@@ -119,6 +127,78 @@
...
+%    \end{macrocode}
+% currently amsmath is loaded in a begindocument hook, so this
+% test is fine. If amstext is loaded earlier (in the kernel), this needs perhaps a change.
+%    \begin{macrocode}
+   \tag_if_active:T
+    {
+      \sys_if_engine_luatex:TF
+       {
+         \def\text@#1{{%
+           \ifcase\mathstyle
```
[the remark this answers is not in the document]
Comment (u-fischer) [reply required/latex-lab/latex-lab-amsmath.dtx:159]: Is there an example where the definition could break?

**Review by josephwright — APPROVED** [review]

File: required/latex-lab/latex-lab-amsmath.dtx (hunk 4 above)
[the remark this answers is not in the document]
Comment (davidcarlisle) [reply required/latex-lab/latex-lab-amsmath.dtx:159]: @u-fischer  if you use `{aaa \over bbb}` then `\mathstyle` will be wrong in aaa you need `\Ustack{aaa \over bbb}`  so luatex scans ahead finds the `\over` and sets `\mathstyle` at the start of the group.  so this requires that `\frac` and any similar command defined by a package uses `\Ustack` in lualatex.

**Pull request merged**